server/executor/testrunner/testrunner_repository.go

```diff
@@ -0,0 +1,149 @@
+package testrunner
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/kubeshop/tracetest/server/pkg/id"
+)
+
+func NewRepository(db *sql.DB) *Repository {
+	return &Repository{db}
+}
+
+type Repository struct {
+	db *sql.DB
+}
+
+func (r *Repository) SetID(testRunner TestRunner, id id.ID) TestRunner {
+	testRunner.ID = id
+	return testRunner
+}
+
+func (r *Repository) Create(ctx context.Context, updated TestRunner) (TestRunner, error) {
+	updated.ID = id.ID("current")
+	return r.Update(ctx, updated)
+}
+
+const (
+	insertQuery = `
+	INSERT INTO test_runners(
+		"id",
+		"name",
+		"required_gates"
+	)
+	VALUES ($1, $2, $3)`
+	deleteQuery = `DELETE FROM test_runners`
+)
+
+func (r *Repository) Update(ctx context.Context, updated TestRunner) (TestRunner, error) {
+	// enforce ID and default
+	updated.ID = "current"
+
+	tx, err := r.db.BeginTx(ctx, nil)
+	defer tx.Rollback()
+	if err != nil {
+		return TestRunner{}, err
+	}
```

Comment:
We have no guarantees that `tx` is not `nil` in an error condition (which can result in a `nil` pointer panic). To avoid that we can swap the error handling, what do you think?

```suggestion
	if err != nil {
		return TestRunner{}, err
	}
	defer tx.Rollback()
```